flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -241,6 +241,18 @@ public Execution getPriorExecutionAttempt(int attemptNumber) {
 		}
 	}
 
+	public Execution getExecutionAttempt(int attemptNumber) {
```

Comment:
i would prefer if you would instead use getCurrentExecutionAttempt if attemptNumber==0 and getPriorExecutionAttempt if attemptNumber>0, for the sole purpose of not having to many methods in the ExecutionVertex class.